internal/services/cdn/frontdoorruleactions/cdn_frontdoor_rule_actions.go

```diff
@@ -354,13 +410,17 @@ func FlattenCdnFrontDoorRouteConfigurationOverrideAction(input cdn.DeliveryRuleR
 			compressionEnabled = config.IsCompressionEnabled == cdn.RuleIsCompressionEnabledEnabled
 			cacheDuration = *config.CacheDuration
 			queryParameters = flattenCsvToStringSlice(config.QueryParameters)
+		} else {
+			cacheBehavior = string(cdn.RuleIsCompressionEnabledDisabled)
 		}
 
 		if override := params.OriginGroupOverride; override != nil {
 			forwardingProtocol = string(override.ForwardingProtocol)
 
-			if group := override.OriginGroup; group != nil && group.ID != nil {
-				originGroupId = *group.ID
+			// NOTE: Need to normalize this ID here because if you modified this in portal the resourceGroup comes back as resourcegroup.
+			// ignore the error here since it was set on the resource in Azure and we know it is valid.
+			if originGroup, err := parse.FrontDoorOriginGroupIDInsensitively(*override.OriginGroup.ID); err == nil {
```

Comment:
in the Expand function we should be parsing these case-sensitively:

```suggestion
			if originGroup, err := parse.FrontDoorOriginGroupID(*override.OriginGroup.ID); err == nil {
```

Reply:
I added this code specifically because if I created the rule in the portal and attempted it import it it would fail because of the casing of the resource group element. Is there something I am missing here?

Reply:
apologies that's my bad for missing this was in the Flatten and not the Expand - we should raise this error if this fails to pass, but this is otherwise fine 👍